src/htmlgen.cpp

```diff
@@ -122,6 +122,32 @@ static void writeServerSearchBox(TextStream &t,const QCString &relPath,bool high
   }
 }
 
+const char *replaceDatetime =  "<span class=\"datetime\"></span>;";
+const char *replaceDate     =  "<span class=\"date\"></span>;";
+const char *replaceTime     =  "<span class=\"time\"></span>;";
+const char *replaceYear     =  "<span class=\"year\"></span>;";
+static void writeBuildDateJS(TextStream &t)
```

Comment:
Since running the script does add to the load time of each page, it is probably wise to only do the string replacements for the elements that are present, and even omit the script if non are present (which looks to be the default).

Reply:
It indeed adds some load time, but it will probably be possible to load only at the moment there are one of these replaceable strings in the header / footer.

I'll have a look at it.

Reply:
Done.

Is there a reason why the `selectBlocks` is done in `substituteHtmlKeywords` for each file instead of doing it directly during the `init` ob the `g_footer` / `g_header`

Reply:
I think there is an even simpler way to do this without any javascript. Here is proof of concept:

a file `date.css` with
```.css
.show-date:after { content: "21-03-2025"; }
```

used like this:
```.html
<!DOCTYPE html>
<html>
<head>
<title>Show variable defined in CSS</title>
<link href="date.css" rel="stylesheet" type="text/css"/>
</head>
<body>
This file was generated on <span class="show-date"></span> from source code.
</body>
</html>
```

Let me know what you think.

Reply:
I followed the way it is done in the ITK project.
I don't know what the time consumption differences between the javascript and css solution would be, but it looks that both approaches are valid approaches,.
I'll leave it up to you to choose the better approach.

Reply:
I've updated the PR and made a CSS version instead of a JS version out of it.